src/object.d

```diff
@@ -4156,15 +4156,19 @@ void __ctfeWrite(scope const(char)[] s) @nogc @safe pure nothrow {}
  * Create RTInfo for type T
  */
 
-template RTInfoImpl(size_t[] pointers)
+template RTInfoImpl(size_t[] pointerBitmap)
 {
-    immutable size_t[pointers.length] data = pointers[];
-    immutable RTInfoImpl = data.ptr;
+    immutable size_t[pointerBitmap.length] RTInfoImpl = pointerBitmap[];
 }
 
 template RTInfo(T)
 {
-    enum RTInfo = RTInfoImpl!(__traits(getPointerBitmap, T));
+    enum pointerBitmap = __traits(getPointerBitmap, T);
+    enum size_t[pointerBitmap.length - 1] allZeros = 0;
+    static if (pointerBitmap[1 .. $] == allZeros)
```

Comment:
I also considered this optimization, but decided against for simplicity of implementation, with less impact on compile times at the cost of a few more bytes emitted. There are not so many likely instantiations of this, though there can also be some extreme examples of gratuitious output.
Not sure if it has a notable impact, but would be good to not create many additional symbols `allZeroes`, e.g. by making it a seperate template.